cmd/sourced/cmd/init.go

```diff
@@ -29,27 +29,27 @@ type initLocalCmd struct {
 }
 
 func (c *initLocalCmd) Execute(args []string) error {
-	reposdir, err := c.reposdirArg()
+	workdirFactory, err := workdir.NewFactory()
```

Comment:
you use `workdirFactory` and `workdirHandler` only once to call either `InitLocal` or `InitOrgs`.
None of those structs have any onther useful methods to be called outside, there is also no reason to pass them as dependencies anywhere.
So it will be simpler if instead of `NewFactory` and `NewHandler`, `workdir` package would just expose `InitLocal` and `InitOrgs`.

Reply:
Done [here](https://github.com/src-d/sourced-ce/pull/183/commits/4738defc7fa7165310c11ed36ddffc523d04a83f).

Reply:
JFYI for the future. It's common and in general a good approach to have structs as you did. But sometimes as in this case you don't need to expose them. So you can make the struct private and expose only methods that operate on it.

Reply:
sure! thanks for the tup!